sample_projects/bin_required_features/Cargo.toml

```diff
@@ -0,0 +1,16 @@
+[package]
+name = "bin_required_features"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+rand = { version = "0.8.5", optional = true }
+
```

Comment:
Can you give the "main" binary a required feature too? Make it one that we *do* pass so we continue building the main binary.

Thus the test would have:

1. Binary with feature we *do pass, binary can build, build is attempted.
2. Binary with feature we *don't* pass, binary can't build, build isn't attempted.